source/develop/release-management/features/virt/enhance-import-export-with-ova.md

```diff
@@ -0,0 +1,139 @@
+
+# Design
+This section details the high-level design of the different aspects described in the previous section.
+
+## Upload OVA
+In principle, uploading an OVA is similar to uploading an image. However, it is more challenging because OVA is a more complex resource than an image.  
+
+The upload process would be implemented differently depending on whether the OVA was generated by oVirt or by other source (and thus requires conversion). In both cases, the user will be able to select a file from within a browser and be presented with the VM configuration that will allow him to configure the parameters for the import process:
+
+1. The user selects an OVA file from within the browser.
+2. The front-end sends the OVF to the back-end.
+3. The back-end parses the OVF and returns back an object that represents the VM.
+4. The front-end displays the VM configuration and allows the user to configure the parameters for the import (namely, VM name, destination storage, and networks).
+
+From this point on, the process depends on the source of the OVA. Note that the source is returned to the front-end by the back-end in the _origin_ field of the VM.  
+
+### Upload OVA generated by oVirt
+If the OVA was generated by oVirt, the rest of the process will be:
+
+1. The back-end adds the VM entity into the database.
+2. The back-end creates the destination volumes for the VM disks using the existing process for this in oVirt.
+3. The back-end requests the imageio proxy to prepare to upload-ova with a list of parameters per disk. Namely, these parameters contain the destination path of the disk and its actual size.
+4. The proxy returns a URL (similar to the upload-image flow).
+5. The back-end passes this URL to the front-end (similar to the upload-image flow).
+6. The front-end starts transmitting the **disks** one by one.
+7. The proxy receives the input stream and streams it to the relevant destination volume.
+
+Note that generating OVA files with compressed disks are out of the scope of this feature. In order to support this, we will need to pass a flag to the proxy that indicates whether or not the disk is compressed (step 3) so the proxy could uncompress the data on-the-fly while streaming it to the destination volume (step 7).
+
+### Upload OVA generated by an external source
+If the OVA was generated by external source, the rest of the process will be:  
+
+1. The back-end adds the VM entity into the database.
+2. The back-end creates a temporary volume to store the OVA in.
+3. The back-end requests (through VDSM) the imagoio proxy to prepare to upload-image-like process.
+4. The proxy returns a URL (similar to the upload-image flow).
+5. The back-end passes this URL to the front-end (similar to the upload-image flow).
+6. The front-end starts transmitting the **whole OVA**.
+7. The proxy streams the input into the created volume.
+8. When the streaming is done, the back-end triggers the existing convert OVA method.
```

Comment:
It needs clarification. You assume raw volumes on ovirt SD. Now that does mean additional space is needed, and additional conversion is needed when the original source is not a raw disk (which in case of VMware it is not). 
Also, we recently enhanced virt-v2v to not require additional copy when input is OVA file. This would be a step back.